types/field_type.go

```diff
@@ -162,18 +180,18 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
 		SetBinChsClnFlag(tp)
 	case int:
 		tp.Tp = mysql.TypeLonglong
-		tp.Flen = len(strconv.FormatInt(int64(x), 10))
+		tp.Flen = lenOfUint64(uint64(abs(int64(x))))
 		tp.Decimal = 0
 		SetBinChsClnFlag(tp)
 	case int64:
 		tp.Tp = mysql.TypeLonglong
-		tp.Flen = len(strconv.FormatInt(x, 10))
+		tp.Flen = lenOfUint64(uint64(abs(x)))
 		tp.Decimal = 0
 		SetBinChsClnFlag(tp)
```

Comment:
Can we merge these two cases?

Reply:
Hmmm... I am very new to golang. Do you mean merge it like `case int64 || int` at the same switch branch?

Reply:
Well, I tried it, but results in more code, so just keep it as is.

Reply:
To merge them  you should use `case int64, int:`
And I think keep them separate make the code consitent with others.